mrjob/conf.py

```diff
@@ -68,8 +67,8 @@ def default_options(self):
     def validated_options(self, opts, error_fmt):
         unrecognized_opts = set(opts) - self.ALLOWED_KEYS
         if unrecognized_opts:
-            log.warn(error_fmt % ', '.join(sorted(unrecognized_opts)))
-            return dict((k, v) for k, v in opts.iteritems()
+            log.warning(error_fmt % ', '.join(sorted(unrecognized_opts)))
```

Comment:
Looks like I'd have to change >100 lines to clean this up, so not in this pull request. Want me to file a "cleanup" issue on it?